src/atomate2/common/jobs/electrode.py

```diff
@@ -0,0 +1,237 @@
+    static_maker: Maker,
+    relax_maker: Maker,
+    get_charge_density: Callable,
+    insertions_per_step: int = 4,
+    n_steps: int | None = None,
+    n_inserted: int = 0,
+) -> Response:
+    """Attempt ion insertion.
+
+    The basic unit for cation insertion is:
+        [get_stable_inserted_structure]:
+            (static) -> N x (chgcar analysis -> relax) -> (return best structure)
+
+    Parameters
+    ----------
+    structure:
+        The structure to insert into.
+    inserted_species:
+        The species to insert.
+    structure_matcher:
+        The structure matcher to use to determine if additional
+        insertion is needed.
+    static_maker:
+        A maker to perform static calculations.
+    relax_maker:
+        A maker to perform relaxation calculations.
+    get_charge_density:
+        A function to get the charge density from a previous calculation.
+        Whether to use the AECCAR0 and AECCAR2 files for the charge density.
+        This is often necessary since the CHGCAR file has spurious effects near the
+        core which often breaks the min-filter algorithms used to identify the local
+        minima.
+    insertions_per_step:
+        The maximum number of ion insertion sites to attempt.
+    n_steps:
+        The maximum number of steps to perform.
+    n_inserted:
+        The number of ions inserted so far, used to help assign a unique name to the
+        different jobs.
+    """
+    if structure is None:
+        return None
+    if n_steps is not None and n_steps <= 0:
+        return None
+    # append job name
+    add_name = f"{n_inserted}"
+
+    static_job = static_maker.make(structure=structure)
+    chg_job = get_charge_density_job(static_job.output.dir_name, get_charge_density)
+    insertion_job = get_inserted_structures(
+        chg_job.output,
+        inserted_species=inserted_element,
+        insertions_per_step=insertions_per_step,
+    )
+    relax_jobs = get_relaxed_job_summaries(
+        structures=insertion_job.output, relax_maker=relax_maker, append_name=add_name
+    )
+
+    min_en_job = get_min_energy_structure(
+        relaxed_summaries=relax_jobs.output,
+        ref_structure=structure,
+        structure_matcher=structure_matcher,
+    )
+    nn_step = n_steps - 1 if n_steps is not None else None
+    next_step = get_stable_inserted_structure(
+        structure=min_en_job.output,
+        inserted_element=inserted_element,
+        structure_matcher=structure_matcher,
+        static_maker=static_maker,
+        relax_maker=relax_maker,
+        get_charge_density=get_charge_density,
+        insertions_per_step=insertions_per_step,
+        n_steps=nn_step,
+        n_inserted=n_inserted + 1,
+    )
+
+    for job_ in [static_job, chg_job, insertion_job, min_en_job, relax_jobs, next_step]:
+        job_.append_name(f" {add_name}")
+
+    replace_flow = Flow(
+        jobs=[static_job, chg_job, insertion_job, relax_jobs, min_en_job, next_step]
```

Comment:
Good point.  I hadn't thought about that since the aggregation pipeline for this stuff is already pretty powerful so most of the analysis and "thinking" was relegated to that and this WF is really just a buck-shot approach to get a bunch of structures into the DB.
But there is good value in getting a document out for some simple applications.  I can add that to the PR.

Reply:
Added some useful document models to the output.
Requires this PR from emmet for serialization to work:

https://github.com/materialsproject/emmet/pull/922